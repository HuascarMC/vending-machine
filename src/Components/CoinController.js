import React from 'react';
import axios from 'axios'

class CoinController extends React.Component {
  render() {

    increaseCoinQuantity(coin) {
      axios.put(`https://vending-machine-server.herokuapp.com/machine/addcoin/${coin}`, { crossdomain: true})
      .then((response) => {
        console.log(response);
      })
      .catch(function (error) {
        console.log(error);
      })
    }

    reduceCoinQuantity(coin) {
      axios.put(`https://vending-machine-server.herokuapp.com/machine/removecoin/${coin}`, { crossdomain: true})
      .then((response) => {
        console.log(response);
      })
      .catch(function (error) {
        console.log(error);
      })
    }

    return(
      <div className="controller-wrapper">
        <ul>
          <li>COKE<button onClick={() => this.increaseCoinQuantity()} className="plus">+</button><button onClick={() => this.reduceCoinQuantity()} className="minus">-</button></li>
          <li>PEPSI<button onClick={() => this.increaseCoinQuantity()} className="plus">+</button><button onClick={() => this.reduceCoinQuantity()} className="minus">-</button></li>
          <li>SODA<button onClick={() => this.increaseCoinQuantity()} className="plus">+</button><button onClick={() => this.reduceCoinQuantity()} className="minus">-</button></li>
          <li>WATER<button onClick={() => this.increaseCoinQuantity()} className="plus">+</button><button onClick={() => this.reduceCoinQuantity()} className="minus">-</button></li>
        </ul>
      </div>
    );
  }
}

export default CoinController;
