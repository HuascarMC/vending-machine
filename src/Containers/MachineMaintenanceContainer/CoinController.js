import React from 'react';
import axios from 'axios';

class CoinController extends React.Component {
  render() {
    return(
        <div className="stock">
          <ul className="stock coin-main">
            <li className="stock-item">DOLL <p className="quantity">{this.props.props.state.DOLLAR}</p><button onClick={() => this.props.props.increaseCoinQuantity("dollar")} className="plus">+</button><button onClick={()=>this.props.props.reduceCoinQuantity("dollar")}className="minus">-</button></li>
            <li className="stock-item">QUAR <p className="quantity">{this.props.props.state.QUARTER}</p><button onClick={() => this.props.props.increaseCoinQuantity("quarter")} className="plus">+</button><button onClick={()=>this.props.props.reduceCoinQuantity("quarter")}className="minus">-</button></li>
            <li className="stock-item">DIME <p className="quantity">{this.props.props.state.DIME}</p><button onClick={() => this.props.props.increaseCoinQuantity("dime")} className="plus">+</button><button onClick={()=>this.props.props.reduceCoinQuantity("dime")}className="minus">-</button></li>
            <li className="stock-item">NICK <p className="quantity">{this.props.props.state.NICKEL}</p><button onClick={() => this.props.props.increaseCoinQuantity("nickel")} className="plus">+</button><button onClick={()=>this.props.props.reduceCoinQuantity("nickel")}className="minus">-</button></li>
            <li className="stock-item">PENN <p className="quantity">{this.props.props.state.PENNY}</p><button onClick={() => this.props.props.increaseCoinQuantity("penny")} className="plus">+</button><button onClick={()=>this.props.props.reduceCoinQuantity("penny")}className="minus">-</button></li>
          </ul>
        </div>

    );
  }
}

export default CoinController;
