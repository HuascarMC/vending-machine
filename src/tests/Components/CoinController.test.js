import React from 'react';
import CoinController from '../../Components/CoinController'
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

var state = {
  dollar: 0,
  quarter: 0,
  dime: 0,
  nickel: 0,
  penny: 0,
}

it('should render without crashing', () => {
  shallow(<CoinController coins={state}/>);
});

it('should render controller-wrapper div', () => {
  const wrapper = shallow(<CoinController coins={state}/>);
  const div = wrapper.find('.controller-wrapper');
  expect(div.exists()).toEqual(true);
});

it('should render u-list', () => {
  const wrapper = shallow(<CoinController coins={state}/>);
  const ul = wrapper.find('ul');
  expect(ul.exists()).toEqual(true);
});

it('should render five list tags', () => {
  const wrapper = shallow(<CoinController coins={state}/>);
  const li = wrapper.find('li');
  expect((li).length).toBe(5);
});

it('should render dollar quantity passed by props', () => {
  const wrapper = shallow(<CoinController coins={state}/>);
  const li = wrapper.find('li').at(0);
  expect(li.text()).toEqual("DOLL 0+-");
});

it('should render quarter quantity passed by props', () => {
  const wrapper = shallow(<CoinController coins={state}/>);
  const li = wrapper.find('li').at(1);
  expect(li.text()).toEqual("QUAR 0+-");
});

it('should render dime quantity passed by props', () => {
  const wrapper = shallow(<CoinController coins={state}/>);
  const li = wrapper.find('li').at(2);
  expect(li.text()).toEqual("DIME 0+-");
});

it('should render nickel quantity passed by props', () => {
  const wrapper = shallow(<CoinController coins={state}/>);
  const li = wrapper.find('li').at(3);
  expect(li.text()).toEqual("NICK 0+-");
});
