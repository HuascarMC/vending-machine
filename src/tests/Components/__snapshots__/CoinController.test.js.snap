// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<div
  className="controller-wrapper"
>
  <ul>
    <li>
      DOLL 
      <button
        className="plus"
        onClick={[Function]}
      >
        +
      </button>
      <button
        className="minus"
        onClick={[Function]}
      >
        -
      </button>
    </li>
    <li>
      QUAR 
      <button
        className="plus"
        onClick={[Function]}
      >
        +
      </button>
      <button
        className="minus"
        onClick={[Function]}
      >
        -
      </button>
    </li>
    <li>
      DIME 
      <button
        className="plus"
        onClick={[Function]}
      >
        +
      </button>
      <button
        className="minus"
        onClick={[Function]}
      >
        -
      </button>
    </li>
    <li>
      NICK 
      <button
        className="plus"
        onClick={[Function]}
      >
        +
      </button>
      <button
        className="minus"
        onClick={[Function]}
      >
        -
      </button>
    </li>
    <li>
      PENN 
      <button
        className="plus"
        onClick={[Function]}
      >
        +
      </button>
      <button
        className="minus"
        onClick={[Function]}
      >
        -
      </button>
    </li>
  </ul>
</div>
`;
